### PR TITLE
Support querying from vLLM server

### PR DESCRIPTION
**Pull request by Chrisytz**
This PR contains two scripts. 

The first one `save_finetuned_model.py` saves the finetuned model's weights as a hugging face model (will integrate this into `save_checkpoint` later).

The second one `query_vllm.py` allows you to perform queries to the vLLM server and returns a readable output.
1. First start the vLLM server ` vllm serve <your model> --max-model-len=<max model length (2048)>`
2. After the server is started run the script like such 
`python query_vllm.py --model=<your model that you started the vLLM server with> "A robe takes 2 bolts of blue fiber and half that much white fiber. How many bolts in total does it take?"`

Also updated the benchmarking script. To test the accuracy of the finetuned model you can pass `--model=<finetuned model>` into the benchmarking script.

### CODE REVIEW

File: benchmarking/save_finetuned_model.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    override_config_kwargs = {
+        "bos_token_id": tokenizer.bos_token_id,
+        "eos_token_id": tokenizer.eos_token_id,
+        "pad_token_id": tokenizer.pad_token_id,
+    }
+    update_model_config(config, override_config_kwargs=override_config_kwargs)
+
+    model = AutoModelForCausalLM.from_pretrained(
+        base_model,
+        torch_dtype=torch.float32,
```
Comment (bhavya01) [inline benchmarking/save_finetuned_model.py:28]: Does this work even if the saved weights dtype is not torch.float32?

Comment (Chrisytz) [reply benchmarking/save_finetuned_model.py:28]: I'm not too sure. I just used the same parameters they used in VERL to load in the pre trained model.
